addons/web/static/src/js/fields/abstract_field.js

```diff
@@ -299,6 +299,32 @@ var AbstractField = Widget.extend({
         return this._render() || $.when();
     },
 
+    /**
+     * add the invalid class on a field
+     */
+    setInvalidClass: function () {
+        this.$el.toggleClass('o_field_invalid',true);
```

Comment:
maybe simpler to use this.$el.addClass('o_field_invalid')

Reply:
done